### DISCUSSION TITLE
[3.3.x] Developer Discussion

### DISCUSSION THREAD

**Discussion by slook** [nicotine-plus/nicotine-plus#2766] [General]
Matters regarding the nicotine-plus code that are relevant for the future version of Nicotine+ 3.3.1

```
**This is not a conversation thread, and as such, please keep replies in-line and on-topic.**
```

**Comment by slook** [comment]
- Comments moved to https://github.com/nicotine-plus/nicotine-plus/discussions/3012#discussioncomment-9462209

**Comment by slook** [comment]
_WONTFIX_ - **Feature suggestion: Highlight the Apply button with in the Preferences dialog when change(s) have been made**

...and I suppose the Cancel button in red destructive colour as well.

Alternatrively, Remove Apply button from preferences dialog as in https://github.com/nicotine-plus/nicotine-plus/discussions/1881#discussioncomment-2914671

**Reply by mathiascode** [reply]
Maybe it's better to keep things as-is. I don't think other applications do this?

**Comment by slook** [comment]
_DONE in Nicotine+ 3.4.0.dev1_ - **Feature suggestion: Highlight any configured Shares in Preferences dialog for which the real path does not exist or cannot be accessed**

When the Shares Not Accessible error dialog is destroyed, then it can be hard to remember which share(s) had the error.

A **`!`** icon or red colour text style would be sufficient for this purpose, that is updated each time the Shares list view is built.

To further expand on this idea, a tooltip or popover could offer some detail about the error(s), such as if a folder cannot be found or perhaps even containing a scrollable list of files which have tag errors or other such problems. Such a dictionary of errors could be managed in Scanner() alongside calls to log.add().

If an icon column is added to the Shares list view, then it would be reassuring to see a green tick icon or similar to indicate the successfully mounted and scanned shared folders without any logged errors.

**Comment by mathiascode** [comment]
_DONE in Nicotine+ 3.4.0.dev1_ - **Remove Text-to-Speech feature in favor of OS screen reader and notifications**

The current Text-to-Speech feature relies on running commands, which isn't really possible in the world of sandboxing, and rather unfriendly for cross-platform users. Nowadays we have OS-level notifications, which at least on GNOME are read by the system's screen reader when enabled. It would be better to use them instead of duplicating effort in our own inferior TTS implementation.

We should verify if different OSes read notifications aloud. If this is the case, I propose we remove the Text-to-Speech feature, and instead add an option to mute specific chats (no notification or sound).

**Comment by slook** [comment]
**Hide irrelevant rooms when finding using text entry in Rooms popover**

Like the browse shares folders tree view search function.

The matching results can be spread wide, there is no find next functionality and it is not obvious that there are additional results further down the list.

**Reply by mathiascode** [reply]
I agree we should restore this behavior at some point. It was removed to make it easier to finish the treeview work in 3.3.0.

**Comment by slook** [comment]
_Resolved in Nicotine+ 3.3.3_ - **Download context menu items missing keyboard accelerators**

Since https://github.com/nicotine-plus/nicotine-plus/commit/5427c5ec212789c6433493df45457f78f3c200bd The D and T keyboard accelerators are gone, perhaps they can be transposed into the remaining menu items instead.

**Reply by slook** [reply]
See PR #2905

**Comment by slook** [comment]
_~DONE~ (REVERTED) in Nicotine+ 3.4.0.dev1_ - **GTK 4 on Windows: Reevaluate when the new GPU renderers are stable**

https://github.com/nicotine-plus/nicotine-plus/blob/13006aedad99c27dc5fc8b0b54af07f79ff58ef1/pynicotine/gtkgui/__init__.py#L131-L134

Update: reverted by https://github.com/nicotine-plus/nicotine-plus/commit/53265a9b3d2b078786755960cff139ec0e1eec16, see https://github.com/nicotine-plus/nicotine-plus/discussions/3012#discussioncomment-13256407

**Comment by slook** [comment]
**Cannot view User Profile, Browse Shares nor Search Files of another user on the same LAN**

Each user has a different port, but is on the same external IP address.

**Reply by mathiascode** [reply]
Are the ports forwarded?

**Reply by slook** [reply]
No, behind public NAT, the DHCP server on the WLAN is a mobile device with a basic WiFi hotspot feature. It automatically assigns addresses in the 192.168.1.0/24 range, that is not configurable. Other local services such as FTP, Samba, Cups, adb, etc, all work after configuring local firewalls.

Software firewall logs show that in the case of Nicotine+ no local connections between machines are ever attempted.

**Reply by mathiascode** [reply]
Ultimately the issue is that the Soulseek network provides the public IP address of users, even ones on the same local network, and all connection attempts are made to it. Unless you forward ports, it's not going to work.

**Reply by slook** [reply]
It isn't that I particularly need clients on the same LAN to see each other (although that would be desirable), the main reason why I think it could be interesting to support local area, or even link-local loopback, connections is because that would allow us to spawn a bunch of clients in a CI testing environment, perhaps each in their own venv or something, alongside a lightweight server instance, in order to run more comprehensive tests for distributed searches, peer connections, file transfers and so on.

> the Soulseek network provides the public IP address of users, even ones on the same local network

That's my point, if the IP address is equal to the peer's but the username is different, then this means the client should ideally try swapping out the peer's address for the local broadcast address to see if there is a nearby client advertising the relevant port.

**Comment by slook** [comment]
_FIXED_ - **Connections count is off by 1 displayed during network outage**

![image](https://github.com/nicotine-plus/nicotine-plus/assets/88614182/5583159f-9e38-46a5-a84b-7d10eb07d4b2)

The count probably includes own user, but that is irrelevant as in Offline state expectation would 0 not 1.

**Reply by mathiascode** [reply]
Fixed in 9c054dde76e5639e519dbc452ff47dcad09c4d07

**Comment by slook** [comment]
**GTK 4 errors printed to console and "Clear All..." button can't recieve touch event**

The Android Oreo 8.1.0 AOSP device screen is only 5" in size, resolution is 1280x720 in forced landscape orientation. Posting these errors here only out of interest for research and development purposes.

Maybe related to high DPI setting and lack of any theme support in the OS, this is probably not a bug in Nicotine+.

```
~/git/nicotine-plus $ python3 ./nicotine
[2024-03-05 11:06:28] Loading Python 3.11.8
[2024-03-05 11:06:28] [Misc] Using Python executable: /data/data/com.termux/files/usr/bin/python3
[2024-03-05 11:06:28] [Misc] Using Nicotine+ executable: /data/data/com.termux/files/home/git/nicotine-plus/pynicotine
[2024-03-05 11:06:28] Loading Nicotine+ 3.3.3.dev1
[2024-03-05 11:06:28] Translation files (.mo) are unavailable, using default English strings
[2024-03-05 11:06:29] Loading GTK 4.12.4
[snip...]
[2024-03-05 11:06:31] Rescanning shares…
[snip...]
(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.469: Allocation width too small. Tried to allocate 90x31, but GtkButton 0x7ddf8fb730 needs at least 116x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.469: Allocation width too small. Tried to allocate 78x25, but GtkBox 0x7ddf9746e0 needs at least 104x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.469: Allocation width too small. Tried to allocate 90x31, but GtkButton 0x7ddf8fb8f0 needs at least 131x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.469: Allocation width too small. Tried to allocate 78x25, but GtkBox 0x7ddf974b60 needs at least 119x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.470: Allocation width too small. Tried to allocate 100x31, but GtkButton 0x7ddf8fbc70 needs at least 114x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.470: Allocation width too small. Tried to allocate 88x25, but GtkBox 0x7ddf9a7460 needs at least 102x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.470: Allocation width too small. Tried to allocate 140x31, but GtkButton 0x7ddf8fbe30 needs at least 195x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.470: Allocation width too small. Tried to allocate 128x25, but GtkBox 0x7ddf9a78e0 needs at least 183x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.588: Allocation width too small. Tried to allocate 90x31, but GtkButton 0x7ddf8fb730 needs at least 116x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.588: Allocation width too small. Tried to allocate 90x31, but GtkButton 0x7ddf8fb8f0 needs at least 131x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.588: Allocation width too small. Tried to allocate 100x31, but GtkButton 0x7ddf8fbc70 needs at least 114x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 11:06:31.589: Allocation width too small. Tried to allocate 140x31, but GtkButton 0x7ddf8fbe30 needs at least 195x31.
[snip...]
[2024-03-05 11:07:26] [Misc] Cannot read color scheme, falling back to GTK theme preference: g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files (2)
[snip...]
(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:27.959: Allocation width too small. Tried to allocate 25x31, but GtkButton 0x7ddf2ec1f0 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:27.960: Allocation width too small. Tried to allocate 13x25, but GtkBox 0x7ddf35ece0 needs at least 27x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:27.960: Allocation width too small. Tried to allocate 30x31, but GtkButton 0x7ddf2ec3b0 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:27.960: Allocation width too small. Tried to allocate 18x25, but GtkBox 0x7ddec56160 needs at least 27x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:28.059: Allocation width too small. Tried to allocate 25x31, but GtkButton 0x7ddf2ec1f0 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:28.059: Allocation width too small. Tried to allocate 30x31, but GtkButton 0x7ddf2ec3b0 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.302: Allocation width too small. Tried to allocate 25x31, but GtkButton 0x7ddfe58370 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.302: Allocation width too small. Tried to allocate 13x25, but GtkBox 0x7ddf84c0e0 needs at least 27x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.303: Allocation width too small. Tried to allocate 30x31, but GtkButton 0x7ddfe58530 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.303: Allocation width too small. Tried to allocate 18x25, but GtkBox 0x7ddf84c560 needs at least 27x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.303: Allocation width too small. Tried to allocate 104x31, but GtkButton 0x7ddfe588b0 needs at least 114x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.303: Allocation width too small. Tried to allocate 92x25, but GtkBox 0x7ddf84ce60 needs at least 102x25.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.398: Allocation width too small. Tried to allocate 25x31, but GtkButton 0x7ddfe58370 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.398: Allocation width too small. Tried to allocate 30x31, but GtkButton 0x7ddfe58530 needs at least 39x31.

(org.nicotine_plus.Nicotine:27483): Gtk-CRITICAL **: 12:18:33.398: Allocation width too small. Tried to allocate 104x31, but GtkButton 0x7ddfe588b0 needs at least 114x31.
[snip...]
```

The application appears to start and operates normally including search and transfer and chat. Hardware video acceleration is as yet unavailable using `termux-x11-nightly` on this machine, which is only usable with `-legacy-drawing`, similar software rendering limitation when using VNC also tested as well and works fine without any lag.

The only exception of the "Clear All..." button in transfers view which cannot be touched by finger event (but can be clicked on with mouse pointer click event), regardless of DPI setting.

Tested with actual finger on the physical screen as well as emulated touch event using mouse in sdk mode via [scrcpy](https://github.com/Genymobile/scrcpy), either via adb USB or adb over WiFi, in both cases this button touch is inoperative, yet all other widgets can be touched okay.

_Physical finger touch event, button can be focused is highlighted but menu doesn't appear:_
![image](https://github.com/nicotine-plus/nicotine-plus/assets/88614182/10f9d07d-78fc-450e-bbcd-090d4a01eab2)

_Emulated touch event via scrcpy, menu doesn't appear until the host Screenshot window takes focus away:_
![image](https://github.com/nicotine-plus/nicotine-plus/assets/88614182/3ead1bc1-46b6-416f-ad62-e90139f7eb93)

_Hardware bluetooth HID mouse directly connected or mouse forwarding over adb via USB pointer click works as normal:_
![image](https://github.com/nicotine-plus/nicotine-plus/assets/88614182/3ff00d63-3484-4d77-b209-eb10e2fae988)

GTK 3.24.8 is also present, and when run with that this problem doesn't exist and menu shows on finger touch fine. Only GTK 4 has issue.

**Reply by mathiascode** [reply]
The popover issue should be https://gitlab.gnome.org/GNOME/gtk/-/issues/6646, fixed in https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/5782. Do you have GTK 4.14.3 available for testing?

**Reply by slook** [reply]
Yes, it was recently upgraded just last week https://github.com/termux/termux-packages/commit/391fec694cba58d6b697a387dbbe4c8e8a4101c3 in PR https://github.com/termux/termux-packages/pull/19986 I will test it soon.

**Reply by mathiascode** [reply]
Did you get a chance to test this?

**Reply by slook** [reply]
I think it did work in the end from what I remember, but I had some other issue with the environment which caused me to give up.

**Comment by slook** [comment]
_FIXED in Nicotine+ 3.3.11_ - **Connection error message during startup**

```
[2024-04-04 04:21:43] [Conn] Cannot send the message over the closed connection: <class 'pynicotine.slskmessages.SharedFoldersFiles'> <S - SharedFoldersFiles> {'folders': 321, 'files': 3970}
[2024-04-04 04:21:43] Connected to server server.slsknet.org:2242, logging in…
[2024-04-04 04:21:44] [Misc] Creating Port Mapping rule...
```

**Reply by mathiascode** [reply]
Some sporadic timing issue when shares initialize and we're also connecting to the server. Only one of them should send a message (ideally), but in this case the message from the share initialization was queued while we were in the process of connecting.

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/7183e2dc25c7ef3de4678b950b60a9aa52884a0f

**Comment by slook** [comment]
_FIXED_ - **Accept private room invitations setting isn't transposed in the roomlist popover when it is changed in the Preferences dialog.**

The `core.chatrooms.request_private_room_toggle()` method only sends the server a message, it doesn't set the value of the toggle.

https://github.com/nicotine-plus/nicotine-plus/commit/1f8c69da72dd882cb7d3f4c0773e2273a65f266c

**Reply by mathiascode** [reply]
I thought I handled that already. I'll take a look.

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/9f7c586dad54b19262720f47d3bb86b41aaaeb43. Seems like I forgot to test it in the other direction.

**Comment by slook** [comment]
_FIXED_ - Non-critical error, console message at runtime:

```
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 969, in on_critical_error
    self._on_critical_error(_exc_type, exc_value, _exc_traceback)
  File "/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 932, in _on_critical_error
    while traceback.tb_next:
AttributeError: 'NoneType' object has no attribute 'tb_next'

Original exception was:
TypeError: song_change() takes 7 positional arguments but 8 were given

```

Program continues running normally, there is no crash.

master at 58543c3d

**Reply by mathiascode** [reply]
Fixed in 2cc1958dfa2683d8da1d44d05a4438e39390fd19

**Comment by slook** [comment]
_ADDED in Nicotine+ 3.3.5_ - **Chat Entry lacks accessible pointer method to execute sending message action**

Older versions of Nicotine+ had clickable icon on the left side of the widget, but it was removed for cosmetic reasons. It should be re-added, it would then match the implementation of the Search Files text entry widget.

Currently the only way to send a message is by pressing the Enter key on a keyboard, but the user might not be using a keyboard to type text, so selected the Enter key action is not always convenient. For example, one may be using a pointing device to open the context menu to Paste text (such as a quote or content from a website), a virtual keyboard, speech-to-text or other type of interface where the Enter key action is not easily accessible.

**Reply by mathiascode** [reply]
> Older versions of Nicotine+ had clickable icon on the left side of the widget, but it was removed for cosmetic reasons.

Which version was this? I don't remember adding/removing an icon there.

Makes sense to add an icon though.

**Reply by mathiascode** [reply]
Not sure which icon would fit here. There aren't any strong contenders in the limited list of standard icons.

**Reply by slook** [reply]
gnome-stock-mail-snd
mail-send-symbolic
mail-send
mail-sent
stock_mail-send
go-last
go-right
next
gtk-goto-last-ltr
gtk-media-play-ltr
media-playback-start
player_start
stock_media-play

**Comment by slook** [comment]
_FIXED_ - **Crash on PageDown key**

```
Nicotine+ Version: 3.3.5.dev3
GTK Version: 3.24.24
Python Version: 3.9.2 (linux)

Type: <class 'AttributeError'>
Value: 'ChatEntry' object has no attribute 'grab_focus_without_selecting'
Traceback:   File "/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/accelerator.py", line 104, in _activate_accelerator
    return self.callback(widget, None, self.user_data)
  File "/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textview.py", line 480, in on_page_down_accelerator
    self.chat_entry.grab_focus_without_selecting()
```

Since https://github.com/nicotine-plus/nicotine-plus/pull/3072 in 3.3.5.dev3

**Comment by slook** [comment]
_FIXED_ - **Warnings on console related to textentry after quit**

```
[2024-08-13 22:33:01] Quit Nicotine+ 3.3.5.dev3!
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:479: Warning: ../../../gobject/gsignal.c:2654: instance '0x2eba310' has no handler with id '2634'
  self.buffer.set_spell_checker(None)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:479: Warning: ../../../gobject/gsignal.c:2696: instance '0x2eba310' has no handler with id '2634'
  self.buffer.set_spell_checker(None)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:483: Warning: ../../../gobject/gsignal.c:2654: instance '0x2eba310' has no handler with id '2634'
  self.entry.set_inline_spell_checking(False)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:483: Warning: ../../../gobject/gsignal.c:2696: instance '0x2eba310' has no handler with id '2634'
  self.entry.set_inline_spell_checking(False)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:479: Warning: ../../../gobject/gsignal.c:2654: instance '0x2ebab20' has no handler with id '7185'
  self.buffer.set_spell_checker(None)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:479: Warning: ../../../gobject/gsignal.c:2696: instance '0x2ebab20' has no handler with id '7185'
  self.buffer.set_spell_checker(None)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:483: Warning: ../../../gobject/gsignal.c:2654: instance '0x2ebab20' has no handler with id '7185'
  self.entry.set_inline_spell_checking(False)
/home/user/Git/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/textentry.py:483: Warning: ../../../gobject/gsignal.c:2696: instance '0x2ebab20' has no handler with id '7185'
  self.entry.set_inline_spell_checking(False)
$
```

Probably regression since PR #3072

**Reply by mathiascode** [reply]
Should be fixed in https://github.com/nicotine-plus/nicotine-plus/commit/cb178c5f344f2940d096885decbcd3a99f87165f.

**Comment by slook** [comment]
_DONE at Nicotine+ 3.3.10_ - **Create `3.3.x` branch**

needed for **_critical_** fixes only, whilst resisting any temptation to backport new features from 3.4.x

Hopefully not much maintenance shall be needed this time, ideally we don't want to repeat what happened in 3.2.x because that was too much work...

... I suggest our backporting policy this time should be that patches are only applied in order to avoid **_Critical Error_** crashes or **_important_** bugs that are considered to be extremely serious.

- This thread will be closed and add new Comments to https://github.com/nicotine-plus/nicotine-plus/discussions/3012

**Comment by mathiascode** [comment]
TODO before stable 3.3.11 release:
- Update country database
- Update copyright years
- Check if release notes are missing anything

**Comment by slook** [comment]
_FIXED in rc2_ - **Gtk-WARNING GtkText - did not receive a focus-out event.**

Nicotine+ 3.3.11rc2
GTK 4.8.3   •   Python 3.11.2

_On disconnected (not userchoice) while window inactive in background (not minimized):_
```
[2026-08-13 01:29:29] Disconnected from server localhost:2242
[2026-08-13 01:29:29] Reconnecting to server in 11 seconds
[2026-08-13 01:29:29] Plugin Debugger: server_disconnect_notification(userchoice=False)
[2026-08-13 01:29:29] [Misc] Backed up and saved file /home/user/.config/nicotine/config

(org.nicotine_plus.Nicotine:3751530): Gtk-WARNING **: 01:29:29.828: GtkText - did not receive a focus-out event.
If you handle this event, you must return
GDK_EVENT_PROPAGATE so the default handler
gets the event as well
[2026-08-13 01:29:40] [Misc] Network interface: lo
[2026-08-13 01:29:40] [Misc] Local IP address: 127.0.0.1
[2026-08-13 01:29:40] [Misc] Maximum number of concurrent connections (sockets): 2048
[2026-08-13 01:29:40] Listening on port: 54321
[2026-08-13 01:29:40] Connecting to localhost:2242
```

_Also, unknown cause:_
```
(org.nicotine_plus.Nicotine:3751530): Gtk-WARNING **: 01:36:53.229: Broken accounting of active state for widget 0x3e9fefb0(GtkPopoverMenu)
```

**Reply by mathiascode** [reply]
First warning is probably the chat entry being disabled when disconnecting? I've seen that warning there before.

Second warning is fixed in newer GTK 4 versions.

**Reply by slook** [reply]
Yes that would be my guess. It's not the first time I've seen it either. The order of events in the disconnecting or reconnecting case seems like it might be slightly incorrect.

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/pull/3855, for reference.

**Comment by slook** [comment]
**Gtk-WARNING gtk_window_set_titlebar() called on a realized window**

On toggling "Use header bar" to enabled in Preferences...
```
[2026-09-21 07:54:29] Loading Python 3.11.2
[2026-09-21 07:54:29] Loading Nicotine+ 3.3.11
...
[2026-09-21 07:54:30] Loading GTK 4.8.3

(org.nicotine_plus.Nicotine:1209695): Gtk-WARNING **: 08:03:52.079: gtk_window_set_titlebar() called on a realized window
```

The warning always appears in the terminal console after toggling the switch on (and pressing Apply) after the setting was previously off.

It doesn't matter which display the Nicotine+ nor anywhere the windows are placed, message is printed every time. No visual glitches other than the console warning. Both windows disappear for about half a second whilst the setting is being applied.

The position of the Preferences dialog is reset to the centre when it reappears after pressing Apply. If pressing OK then the main window becomes hidden so for some reason it is necessary to unminimize it after changing the setting either way.

```
System:
  Desktop: Xfce v: 4.20.0 tk: Gtk v: 3.24.38 info: xfce4-panel wm: xfwm v: 4.20.0 vt: 7
    dm: LightDM v: 1.32.0 Distro: MX-23.6_x64 Libretto September 15  2024 base: Debian GNU/Linux 12
    (bookworm)
Graphics:
  Device-1: Intel 2nd Generation Core i7 Processor Family Integrated Graphics vendor: Lenovo
    driver: i915 v: kernel arch: Gen-6 code: Sandybridge process: Intel 32nm built: 2011 ports:
    active: LVDS-1,VGA-1 empty: none bus-ID: 00:02.0 chip-ID: 8086:0126 class-ID: 0300
  Device-2: NVIDIA GF119M [Quadro NVS 4200M] vendor: Lenovo driver: N/A alternate: nouveau
    non-free: series: 390.xx+ status: legacy-active (EOL~late 2022) arch: Fermi code: GF1xx
    process: 40/28nm built: 2010-16 pcie: gen: 1 speed: 2.5 GT/s lanes: 16 bus-ID: 01:00.0
    chip-ID: 10de:1057 class-ID: 0300
  Display: x11 server: X.Org v: 1.21.1.7 compositor: xfwm v: 4.20.0 driver: X:
    loaded: modesetting unloaded: fbdev,vesa dri: crocus gpu: i915 display-ID: :0.0 screens: 1
  Screen-1: 0 s-res: 3520x1080 s-dpi: 96 s-size: 932x286mm (36.69x11.26") s-diag: 975mm (38.38")
  Monitor-1: LVDS-1 pos: primary,bottom-r model: LG Display 0x02e2 built: 2010 res: 1600x900
    hz: 60 dpi: 131 gamma: 1.2 size: 310x174mm (12.2x6.85") diag: 355mm (14") ratio: 16:9
    modes: 1600x900
  Monitor-2: VGA-1 pos: top-left model: Lenovo LEN S22e-19 serial: <filter> built: 2019
    res: 1920x1080 hz: 60 dpi: 102 gamma: 1.2 size: 476x268mm (18.74x10.55") diag: 546mm (21.5")
    ratio: 16:9 modes: max: 1920x1080 min: 720x400
  API: OpenGL v: 3.3 Mesa 22.3.6 renderer: Mesa Intel HD Graphics 3000 (SNB GT2)
    direct-render: Yes
```
